fdbserver/OldTLogServer_4_6.actor.cpp

```diff
@@ -412,7 +414,9 @@ namespace oldTLog_4_6 {
 				// These are initialized differently on init() or recovery
 				recoveryCount(), stopped(false), initialized(false), queueCommittingVersion(0), newPersistentDataVersion(invalidVersion), recovery(Void())
 		{
-			startRole(Role::TRANSACTION_LOG,interf.id(), UID());
+			std::map<std::string, std::string> details;
+			details["TLogVersion"] = TLOG_VERSION;
```

Comment:
Hmm, this is probably better, though it's not enough to point me to which source file is being run (which is mainly why I'm adding this). We could also add the spill type, which I think gives us all the info we need.

Reply:
Ah, maybe this is suggesting that the version is fixed for each implementation rather than using the configured version, which due to other factors may map to different implementations. If so, then I think that works. The only awkward thing currently is that `V1` is commented out in the enum but would probably be what we want in the 4.6 implementation. I can comment it in and use it if there's no objection.

Reply:
In my original version, I put the `#define` where the namespace declaration is or would be so that hopefully it would get updated when a new version needed to be created. Any thoughts on how to ensure that with this new approach?

Reply:
Ah, fair point.  That's going to be a bit complicated overall then.
TLogVersion will indeed not uniquely identify a file, as on release-6.2,
TLogSpillType is needed to decide between 6_0 and TLogServer.  (TLogVersion,
TLogSpillType) is enough to tell you which file/implementation is being used if
you're willing to read TLogFnForOptions.  If you're explicitly just looking for
the filename, you could just use a bit of strrchr magic on top of __FILE__ to
be super copy-paste defensive, but this is also fine.

Note that on `master`, the code you're touching has changed a bit, and some of
the situation has changed.  I realized that unifying the spilling types into
one file was actually reasonably simple (and I regret not realizing and doing
this earlier), so a SharedTLog can now contain a mix of value and reference
spilled TLogs.  This probably means that even if you logged out the file, when
a TLogData is made, you'll also want to log out the TLogSpillType when a
LogData is created.

In the end, logging the file and TLogVersion from SharedTLog/TLogData and TLogSp
illType from LogData will probably be the most comprehensive and correct thing t
o do (on master).  I'm fine with just logging File on the 6.2 branch, and then a
dding TLogVersion and TLogSpillType on master sometime later/for the 7.0 branch.

I've maybe finally pushed `protocolVersion` into enough things that I can stop
having to fork the TLog code...

Reply:
I've opted to just remove this field for now, though I added an unrelated one in its place.